scipy/_lib/_numpy_compat.py

```diff
@@ -45,3 +49,236 @@ def _assert_warns(warning_class, func, *args, **kw):
                 raise AssertionError("First warning for %s is not a "
                         "%s( is %s)" % (func.__name__, warning_class, l[0]))
         return result
+
+
+if NumpyVersion(np.__version__) >= '1.12.0.dev':
```

Comment:
and of course that also provides robustness about ``numpy`` not actually including this - ``__numpy_ufunc__`` lesson:(